AEPAssurance/Source/AssuranceSessionOrchestrator.swift

```diff
@@ -23,6 +23,8 @@ class AssuranceSessionOrchestrator: AssurancePresentationDelegate, AssuranceConn
     /// A buffer for holding the events until the initial Assurance session associated with
     /// the app launch happens. This is emptied once a session has been connected.
     var outboundEventBuffer: ThreadSafeArray<AssuranceEvent>?
+    
+    let orchestratorQueue = DispatchQueue(label: "com.adobe.assurance.orchestrationQueue")
```

Comment:
Are you protecting the `session` access from different threads using this queue? If so, there are a lot of other places where you access session outside the queue.

Reply:
I am only wanting to protect the session and outboundEventBuffer variables under one specific scenario, which is if an event is queued as a session is being terminated. This is also done in Android by using `synchronized` methods for `createSession` and `terminateSession`. See [here](https://github.com/adobe/aepsdk-assurance-android/blob/main/code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/AssuranceSessionOrchestrator.java#L229). I did not want to protect ALL cases of the session variable because it would be a large refactor, for something which we have not had actual customer issues for.